### PR TITLE
Implement event emission, add type checking

### PR DESCRIPTION
**Pull request by impl**

### CODE REVIEW

File: support/python/src/nebula_sdk/client.py [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
-        (_, _, path, query, fragment) = urlsplit(request.url)
+    def send(self, request: PreparedRequest, stream: bool = False,
+             timeout: Optional[Union[
+                 float, Tuple[float, float], Tuple[float, None]]
+             ] = None,
+             verify: Union[bool, str] = True,
+             cert: Optional[Union[
+                 bytes, Text, Container[Union[bytes, Text]]]
+             ] = None,
+             proxies: Optional[Mapping[str, str]] = None) -> Response:
```
Comment (bradhe) [inline support/python/src/nebula_sdk/client.py:27]: This is horribly complicated. Can it be encapsulated in it's own type? Is that something you considered? Given where we're at right now I think we can probably save that for another day but just a thought.

File: support/python/tests/test_events.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    def test_emit(self, requests_mock: Adapter) -> None:
+        requests_mock.register_uri(
+            'POST', 'http+api://api/events',
+            text='OK',
+            request_headers={'content-type': 'application/json'},
+            additional_matcher=lambda request:
+                request.json() == {'data': {'foo': 'bar'}},
+        )
+        Events(new_session(api_url='http://api')).emit({'foo': 'bar'})
```
Comment (bradhe) [inline support/python/tests/test_events.py:16]: this test begs the question...what behavior do we expect when this fails? and how do we expect our system/our users to handle that? it's an edge case, but should be considered as part of the "logging" topic.

Comment (impl) [reply support/python/tests/test_events.py:16]: Most (all?) upstream providers retry webhooks for this reason, although we should probably make sure our internal retry logic for these requests is sound too.
